ValveKeyValue/ValveKeyValue/KVSerializerOptions.cs

```diff
@@ -1,20 +1,17 @@
-﻿namespace ValveKeyValue
+﻿using System.Collections.Generic;
+using System.Runtime.InteropServices;
+
+namespace ValveKeyValue
 {
     /// <summary>
     /// Options to use when deserializing a KeyValues file.
     /// </summary>
     public sealed class KVSerializerOptions
     {
-        string[] conditions;
-
         /// <summary>
         /// Gets or sets a list of conditions to use to match conditional values.
         /// </summary>
-        public string[] Conditions
-        {
-            get { return conditions ?? new string[0]; }
-            set { conditions = value; }
-        }
+        public IList<string> Conditions { get; set; } = new List<string>();
```

Comment:
I don't like having initialization code both inline `{ get; set; } = ` and in the constructor, particularly when one depends on the behaviour/order of the other.

Reply:
Yeah I don't know what's the sanest way of implementing this, feel free to take over.

Reply:
See #29.